iotilecore/iotile/core/hw/debug/flash_board_step.py

```diff
@@ -0,0 +1,22 @@
+from __future__ import (unicode_literals, print_function, absolute_import)
+from builtins import str
+from iotile.core.hw.hwmanager import HardwareManager
+from iotile.core.exceptions import ArgumentError
+
+class FlashBoardStep (object):
+    def __init__(self, args):
+        if(args.get('file') is None):
+            raise ArgumentError("PromptStep Parameter Missing", parameter_name='file')
+        if(args.get('port') is None):
+            raise ArgumentError("PromptStep Parameter Missing", parameter_name='port')
+
+        if 'jlink' not in args['port'][0:5]:
```

Comment:
This will fail if `len(args['port']) < 5`.  You should write as `if not args['port'].startswith('jlink')`